x-pack/platform/plugins/shared/index_management/public/application/sections/home/index_list/index_table/doc_count.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { EuiLoadingSpinner } from '@elastic/eui';
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { INTERNAL_API_BASE_PATH } from '../../../../../../common/constants';
+
+interface DocCountCellProps {
+  indexName: string;
+  httpSetup: HttpSetup;
+}
+
+export const DocCountCell = ({ indexName, httpSetup }: DocCountCellProps) => {
+  const [count, setCount] = React.useState<number>();
+  const [countError, setCountError] = React.useState<boolean>(false);
+
+  useEffect(() => {
```

Comment:
If we start a new request we need to reset the state so that spinner/error states can work as expected on each rerun without staleness

```suggestion
  useEffect(() => {
  setCount(undefined)
  setCountError(false)
```

Reply:
same answer as https://github.com/elastic/kibana/pull/247478#discussion_r2648491599